provider/oracle/environ.go

```diff
@@ -0,0 +1,389 @@
+	// if not, change it to true
+	// if the key is not present in the oracle infrstracture then we should
+	// make a request and upload it to make use for further bootstraping the
+	// controller or adding other machines
+	nameKey, err := uploadSSHControllerKeys(o.p.client, keys[0])
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	instance, err := launchBootstrapConstroller(o.p.client, oci.InstanceParams{
+		Relationships: nil,
+		Instances: []oci.Instances{
+			{
+				Shape:     spec.InstanceType.Name,
+				Imagelist: imagelist,
+				Name:      o.cfg.Name(),
+				Label:     o.cfg.Name(),
+				SSHKeys:   []string{nameKey},
+				Hostname:  o.cfg.Name(),
+				Tags:      []string{o.cfg.Name()},
+				// TODO(sgiulitti): add here the userdata
+				Attributes:  nil,
+				Reverse_dns: false,
+				// TODO(sgiulitti): make vm generate a public address
+			},
+		},
+	})
+
+	result := &environs.StartInstanceResult{
+		Instance: instance,
+	}
+
+	return result, nil
+}
+
+// StopInstances shuts down the instances with the specified IDs.
+// Unknown instance IDs are ignored, to enable idempotency.
+func (o oracleEnviron) StopInstances(...instance.Id) error {
+	return nil
+}
+
+// AllInstances returns all instances currently known to the broker.
+func (o oracleEnviron) AllInstances() ([]instance.Instance, error) {
+	resp, err := o.p.client.AllInstances()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	if len(resp.Result) == 0 {
+		return nil, environs.ErrNoInstances
+	}
+
+	all := make([]instance.Instance, 0, len(resp.Result))
+	for _, val := range resp.Result {
+		inst, err := newInstance(&val)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		all = append(all, inst)
+	}
+
+	return all, nil
+}
+
+// MaintainInstance is used to run actions on jujud startup for existing
+// instances. It is currently only used to ensure that LXC hosts have the
+// correct network configuration.
+func (o oracleEnviron) MaintainInstance(args environs.StartInstanceParams) error {
+	return nil
+}
+
+// Config returns the configuration data with which the Environ was created.
+// Note that this is not necessarily current; the canonical location
+// for the configuration data is stored in the state.
+func (o oracleEnviron) Config() *config.Config {
+	return o.cfg
+}
+
+// ConstraintsValidator returns a constraints.Validator instance which
+// is used to validate and merge constraints.
+//
+// Validator defines operations on constraints attributes which are
+// used to ensure a constraints value is valid, as well as being able
+// to handle overridden attributes.
+//
+// This will use the default validator implementation from the constraints package.
+func (o oracleEnviron) ConstraintsValidator() (constraints.Validator, error) {
+	// list of unsupported oracle provider constraints
+	unsupportedConstraints := []string{
+		constraints.Container,
+		constraints.CpuPower,
+		constraints.RootDisk,
+		constraints.Arch,
```

Comment:
I don't think we should have it unsupported; just register a vocab that includes only "amd64".